vision_agent_tools/tools/text_to_object_detection.py

```diff
@@ -0,0 +1,42 @@
+from typing import List, Any
+from PIL import Image
+from pydantic import BaseModel
+from vision_agent_tools.shared_types import BaseTool
+from vision_agent_tools.models.model_registry import get_model
+
+
+class TextToObjectDetectionOutput(BaseModel):
+    output: Any
+
+
+class TextToObjectDetection(BaseTool):
+    """
+    Tool to perform object detection based on text prompts using a specified ML model
+    """
+
+    def __init__(self, model: str):
```

Comment:
I think it's ok to merge like this, but we should not have model be a generic string. It should be an string `Enum` that defines specific options for each task. For example, `InternLMXComposer2` cannot run this task, but `FlorenceV2` can